keps/sig-api-machinery/4020-unknown-version-interoperability-proxy/README.md

```diff
@@ -240,50 +241,28 @@ This might be a good place to talk about core concepts and how they relate.
+    2. an informer cache of resources served by each peer apiserver in the cluster(Remote Discovery cache). This cache will be updated by an informer on [apiserver identity Lease objects](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1965-kube-apiserver-identity/README.md#proposal). The informer's event handler will make remote discovery calls to peer apiservers whose lease objects are added or updated. 
 
 2. This filter will pass on the request to the next handler in the local aggregator chain, if:
    1. It is a non resource request
-   2. The StorageVersion informer cache hasn't synced yet or if `StorageVersionManager.Completed()` has returned false. We will serve error 503 in this case
+   2. The Local Discovery cache or the Remote Discovery cache hasn't synced yet. We will serve error 503 in this case
    3. The request has a header `X-Kubernetes-APIServer-Rerouted:true` that indicates that this request has been proxied once already. If for some reason the resource is not found locally, we will serve error 503
-   4. No StorageVersion was retrieved for it, meaning the request is for an aggregated API or for a custom resource
-   5. If the local apiserver ID is found in the list of serviceable-by server IDs from the internal map
+   4. The requested resource was listed in the Local Discovery cache
+   5. No other peer apiservers were found to exist in the cluster
 
-3. If the local apiserver ID is not found in the list of serviceable-by server IDs, a random apiserver ID will be selected from the retrieved list and the request will be proxied to this apiserver
+3. If the requested resource was not found in the Local Discovery cache, we will try to fetch the resource from the Remote Discovery cache. The request will then be proxied to any peer apiserver thats found to be able to serve the resource as indicated in the Remote Discovery cache. 
 
-4. If there is no apiserver ID retrieved for the requested GVR, we will serve 404 with error `GVR <group_version_resource> is not served by anything in this cluster`
+4. If there is no eligible apiserver found in the Remote Discovery cache for the requested resource, we will serve 404 with error `GVR <group_version_resource> is not served by anything in this cluster`
 
 5. If the proxy call fails for network issues or any reason, we serve 503 with error `Error while proxying request to destination apiserver`
 
-6. We will also add a poststarthook for the apiserver to ensure that it does not start serving requests until we are done creating/updating SV objects
+6. We will also add a poststarthook for the apiserver to ensure that it does not start serving requests until we have populated both Local Discovery and Remote Discovery caches.
```

Comment:
+1 on local discovery poststarthook. Remote discovery could be considerably slower? Do we really want to block on those as well? When working on aggregated discovery, we specifically did not block startup on remote apiservices (which is slightly different than remote apiservers) to avoid adding excessive startup time.

Reply:
Hmm, the other alternative is serve 503 if the peer-agg-discovery cache hasnt yet synced. I guess we could do that if this(adding delay in apiserver bootstrap) is a major concern ?

Reply:
Modified to say that the poststarthook will block traffic-serving until the local discovery cache is full and the apiserver identity informer is synced